streams/src/main/java/org/apache/kafka/streams/kstream/internals/KGroupedStreamImpl.java

```diff
@@ -188,6 +191,137 @@ public Long apply(K aggKey, V value, Long aggregate) {
                 storeSupplier);
     }
 
+    @SuppressWarnings("unchecked")
+    @Override
+    public <T> KTable<Windowed<K>, T> aggregate(final Initializer<T> initializer,
+                                                final Aggregator<? super K, ? super V, T> aggregator,
+                                                final Merger<? super K, T> sessionMerger,
+                                                final SessionWindows sessionWindows,
+                                                final Serde<T> aggValueSerde,
+                                                final String storeName) {
+        Objects.requireNonNull(storeName, "storeName can't be null");
+        return aggregate(initializer,
+                         aggregator,
+                         sessionMerger,
+                         sessionWindows,
+                         aggValueSerde,
+                         storeFactory(keySerde, aggValueSerde, storeName)
+                          .sessionWindows(sessionWindows.maintainMs()).build());
+
+
+    }
+
+    @SuppressWarnings("unchecked")
+    @Override
+    public <T> KTable<Windowed<K>, T> aggregate(final Initializer<T> initializer,
+                                                final Aggregator<? super K, ? super V, T> aggregator,
+                                                final Merger<? super K, T> sessionMerger,
+                                                final SessionWindows sessionWindows,
+                                                final Serde<T> aggValueSerde,
+                                                final StateStoreSupplier<SessionStore> storeSupplier) {
+        Objects.requireNonNull(initializer, "initializer can't be null");
+        Objects.requireNonNull(aggregator, "aggregator can't be null");
+        Objects.requireNonNull(sessionWindows, "sessionWindows can't be null");
+        Objects.requireNonNull(sessionMerger, "sessionMerger can't be null");
+        Objects.requireNonNull(storeSupplier, "storeSupplier can't be null");
+
+        return (KTable<Windowed<K>, T>) doAggregate(
```

Comment:
It is consistent with what is already there. Also, i can't think of another way of doing it without duplicating `doAggregate`, which i'd rather not do